leases/schema.py

```diff
@@ -0,0 +1,58 @@
+import graphene
+from django.utils.translation import ugettext_lazy as _
+from graphene_django import DjangoConnectionField, DjangoObjectType
+from graphql_jwt.decorators import login_required, superuser_required
+
+from berth_reservations.exceptions import VenepaikkaGraphQLError
+from resources.schema import BerthNode
+
+from .enums import LeaseStatus
+from .models import BerthLease
+
+LeaseStatusEnum = graphene.Enum.from_enum(LeaseStatus)
+
+
+class BerthLeaseNode(DjangoObjectType):
+    berth = graphene.Field(BerthNode)
+    status = LeaseStatusEnum()
+
+    class Meta:
+        model = BerthLease
+        interfaces = (graphene.relay.Node,)
+
+    @classmethod
+    @login_required
+    def get_node(cls, info, id):
+        node = super().get_node(info, id)
+        if not node:
+            return None
+
+        user = info.context.user
+        # TODO: implement proper permissions
+        if (
+            hasattr(node, "customer")
```

Comment:
the attributes will be there, but they might be `None`. thus, IMO a more clear check is:
```python
if (
    node.customer
    and node.customer.user
    and node.customer.user == user
) or user.is_superuser:
    return node
```

Reply:
🤔 fair point, I was doing it like that because it was failing because it didn't have a `user` attr, but I think I should've checked for `node.customer` instead of `node.user` :derp: